### ISSUE TITLE
Development of v0.0.4

### ISSUE THREAD

**Issue by sketchpunk** [sketchpunklabs/ossos#2]
**[ Main Focus ]**
- Animation
  - Blending Animations
  - Learn how to normalize time for smooth blending
  - Figure out how to do phase matching ( https://twitter.com/nan2cc/status/1266792167206338560 )
  - Maybe a framework to handle Blend Trees?
  - Export the results of IK Animation Retargeting. So no need to do realtime ik retargeting all the time.
- Character controller ( get things to actually move )

**[ Secondary ]**
- Bone Springs
  - Not happy with new version, will revisit old prototype version
  - Try to handle Implicit Euler & regular spring force equations, maybe treat them as modes or something.

**[ Maybe? ]**
- Make Playcanvas examples

**[ Most Likely ]**
- Find & Make usable more models to play with.
- Try to model my own character

**Comment by sbiaudet** [comment]
Hi, do you have any release date for this 0.0.4 version ?

**Comment by sketchpunk** [comment]
> Hi, do you have any release date for this 0.0.4 version ?

Not really. I made the decided to rewrite the library since the API felt hard to use ( made bad decisions). Also the math library dependency I had complicated things, so ossos will have its own mini math lib builtin to make it easier to code since the project is very math heavy.

0.0.4 is in the ossos_next branch. I've rewritten big chunks of things along with lots of addition. The whole animation part has more features then v0.0.3. Been slowly tinkering with how the IK Rig Api should be written & trying to build some kind of "Control Rig" to make it easy to let the user pose characters on the web.

https://github.com/sketchpunklabs/ossos/tree/ossos_next
